src/Toolkit/Toolkit/UI/LayerContentDataSource.cs

```diff
@@ -362,10 +362,10 @@ private IEnumerable<Action> TrackLayerContentsRecursive(IEnumerable<ILayerConten
                 {
                     if (layer is INotifyPropertyChanged inpc)
                     {
-                        var listener = new Internal.WeakEventListener<INotifyPropertyChanged, object?, PropertyChangedEventArgs>(inpc)
+                        var listener = new Internal.WeakEventListener<LayerContentDataSource<T>, INotifyPropertyChanged, object?, PropertyChangedEventArgs>(this, inpc)
                         {
-                            OnEventAction = (instance, source, eventArgs) => Layer_PropertyChanged((ILayerContent)instance, eventArgs.PropertyName),
-                            OnDetachAction = (instance, weakEventListener) => instance.PropertyChanged -= weakEventListener.OnEvent,
+                            OnEventAction = static (instance, source, eventArgs) => instance?.Layer_PropertyChanged(source as ILayerContent, eventArgs.PropertyName),
+                            OnDetachAction = static (instance, source, weakEventListener) => (source as INotifyPropertyChanged).PropertyChanged -= weakEventListener.OnEvent,
```

Comment:
Is this cast necessary? I expect `source` to already be of type INotifyPropertyChanged here.